bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolder.java

```diff
@@ -3470,6 +3471,17 @@ public void setSelectionForeground (Color color) {
 	if (selectedIndex > -1) redraw();
 }
 
+/**
+ * Sets the thickness of the highlight bar on the selected tab.
+ *
+ * @param thickness the desired thickness. Set to {@code 0} to not draw a highlight bar
+ * @since 3.121
+ */
+public void setSelectionBarThickness(int thickness) {
```

Comment:
What are the allowed values for thickness? May be a value between 0 to 3? We should check if thickness is within the allowed range. Currently user code can specify any value including negative values or large values.

Reply:
Any positive value should be fine.

Reply:
The allowed values are still not clear. CTabFolderRenderer#ITEM_TOP_MARGIN is 2 so going by javadoc they values can be only 0 and 1, but I think that's not intended. It's better to sanity check the thickness (especially for negative and large values) to prevent unexpected behavior.